File: shared/Models/MatchStatus.swift
import Foundation

public enum MatchStatus: String, Codable, Sendable, Equatable {
    case inProgress
    case completed
    case endedEarly
    case discarded

    public var isTerminal: Bool {
        switch self {
        case .inProgress:
            return false
        case .completed, .endedEarly, .discarded:
            return true
        }
    }

    public var displayName: String {
        switch self {
        case .inProgress: return "In Progress"
        case .completed: return "Completed"
        case .endedEarly: return "Ended Early"
        case .discarded: return "Discarded"
        }
    }
}

File: android/domain/src/main/kotlin/com/padelscore/domain/MatchStatus.kt
package com.padelscore.domain

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class MatchStatus {
    @SerialName("inProgress")
    InProgress,

    @SerialName("completed")
    Completed,

    @SerialName("endedEarly")
    EndedEarly,

    @SerialName("discarded")
    Discarded,
    ;

    val isTerminal: Boolean
        get() = this != InProgress

    val displayName: String
        get() = when (this) {
            InProgress -> "In Progress"
            Completed -> "Completed"
            EndedEarly -> "Ended Early"
            Discarded -> "Discarded"
        }
}
